File: onda_transversal.py
# coding=utf-8
from numpy import * 
from scipy import * 
from matplotlib.pyplot import *
from matplotlib import animation

def data_gen(framenumber, p, plotxp):
    p = sin( k*x - w*t[framenumber] )
    ax.clear()
    plot(x,p,'ko')
    stem(x,p,'k--')
    xlabel(r'$X$',fontsize=18)
    ylabel(r'$Y$',fontsize=18)
    grid(True)
    return plotxp

quadros = 101
nx    = 21
w     = 1.0
k     = 1.0
xmax  = 2.0*pi
t     = linspace(0.0,20.0*pi,quadros)
x     = linspace(0.0,xmax,nx)

# Definir parâmetros iniciais:
p = sin( k*x - w*t[0] )
fig = figure(1)
ax = fig.gca()
plotxp = plot(x,p,'ko')

# Animação:
animacao = animation.FuncAnimation(fig, 
data_gen, fargs=(p, plotxp),
frames=quadros,interval=1, blit=False)

show()

File: onda_transversal.m
% Fundamentos de Acústica Submarina 

clear all, close all 

quadros = 101;
nx    = 21;
w     = 1.0;
k     = 1.0;
xmax  = 2.0*pi;
t     = linspace(0.0,20.0*pi,quadros);
x     = linspace(0.0,xmax,nx);

for i = 1:quadros
p = sin( k*x - w*t(i) );
figure(1)
plot(x,p,'ko'), hold on
stem(x,p,'k--'), hold off
xlabel('X','FontSize',18)
ylabel('Y','FontSize',18)
box on, grid on
pause(0.1)
endfor

